src/Types/Type.php

```diff
@@ -86,6 +86,23 @@ public static function paginatedList(BaseType|string $wrappedType): ModelPaginat
         return $type;
     }
 
+    public static function edge(Relation $relation): ModelEdge|ModelEdgePaginator
+    {
+        $type = new ModelEdge($relation);
+        if (!isset(static::$types[$type->name])) {
+            static::$types[$type->name] = $type;
+        }
+        $type = static::$types[$type->name];
+        if ($relation->type === Relation::BELONGS_TO_MANY || $relation->type === Relation::HAS_MANY) {
+            $type = new ModelEdgePaginator($type);
+            if (!isset(static::$types[$type->name])) {
+                static::$types[$type->name] = $type;
+            }
+            $type = static::$types[$type->name];
+        }
```

Comment:
wird ganz schön oft `type` deklariert, bissl difficult lesbar

Reply:
ich hätte den plan (in einem zukünftigen PR) die ganzen zugriffe auf `static::$types` in ne eigene methode auszulagern - einfach schon deshalb, weil das inzwischen relativ oft passiert - auch in anderen methoden.